controllers/components/datasciencepipelines/datasciencepipelines.go

```diff
@@ -0,0 +1,101 @@
+		"IMAGESV2_ARGO_ARGOEXEC":           "RELATED_IMAGE_ODH_DATA_SCIENCE_PIPELINES_ARGO_ARGOEXEC_IMAGE",
+		"IMAGESV2_ARGO_WORKFLOWCONTROLLER": "RELATED_IMAGE_ODH_DATA_SCIENCE_PIPELINES_ARGO_WORKFLOWCONTROLLER_IMAGE",
+		"V2_DRIVER_IMAGE":                  "RELATED_IMAGE_ODH_ML_PIPELINES_DRIVER_IMAGE",
+		"V2_LAUNCHER_IMAGE":                "RELATED_IMAGE_ODH_ML_PIPELINES_LAUNCHER_IMAGE",
+		"IMAGESV2_ARGO_MLMDGRPC":           "RELATED_IMAGE_ODH_MLMD_GRPC_SERVER_IMAGE",
+	}
+
+	if err := deploy.ApplyParams(DefaultPath, imageParamMap); err != nil {
+		return fmt.Errorf("failed to update images on path %s: %w", DefaultPath, err)
+	}
+
+	return nil
+}
+
+func GetComponentCR(dsc *dscv1.DataScienceCluster) *componentsv1.DataSciencePipelines {
+	dataSciencePipelinesAnnotations := make(map[string]string)
+
+	switch dsc.Spec.Components.DataSciencePipelines.ManagementState {
+	case operatorv1.Managed, operatorv1.Removed:
+		dataSciencePipelinesAnnotations[annotations.ManagementStateAnnotation] = string(dsc.Spec.Components.DataSciencePipelines.ManagementState)
+	default: // Force and Unmanaged case for unknown values, we do not support these yet
+		dataSciencePipelinesAnnotations[annotations.ManagementStateAnnotation] = "Unknown"
+	}
+
+	return &componentsv1.DataSciencePipelines{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       componentsv1.DataSciencePipelinesKind,
+			APIVersion: componentsv1.GroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        componentsv1.DataSciencePipelinesInstanceName,
+			Annotations: dataSciencePipelinesAnnotations,
+		},
+		Spec: componentsv1.DataSciencePipelinesSpec{
+			DataSciencePipelinesCommonSpec: dsc.Spec.Components.DataSciencePipelines.DataSciencePipelinesCommonSpec,
+		},
+	}
+}
+
+func UnmanagedArgoWorkFlowExists(ctx context.Context, cli client.Client) error {
+	workflowCRD := &apiextensionsv1.CustomResourceDefinition{}
+	if err := cli.Get(ctx, client.ObjectKey{Name: ArgoWorkflowCRD}, workflowCRD); err != nil {
+		if k8serr.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("failed to get existing Workflow CRD : %w", err)
+	}
+	// Verify if existing workflow is deployed by ODH with label
+	odhLabelValue, odhLabelExists := workflowCRD.Labels[labels.ODH.Component(componentsv1.DataSciencePipelinesComponentName)]
+	if odhLabelExists && odhLabelValue == "true" {
+		return nil
+	}
+
+	return fmt.Errorf("%s CRD already exists but not deployed by this operator. "+
+		"Remove existing Argo workflows or set `spec.components.datasciencepipelines.managementState` to Removed to proceed ", ArgoWorkflowCRD)
+}
+
+func SetExistingArgoCondition(conditions *[]conditionsv1.Condition, reason, message string) {
+	status.SetCondition(conditions, string(status.CapabilityDSPv2Argo), reason, message, corev1.ConditionFalse)
+	status.SetComponentCondition(conditions, componentsv1.DataSciencePipelinesComponentName, status.ReconcileFailed, message, corev1.ConditionFalse)
+}
```

Comment:
@lburgazzoli I added that here

Reply:
but that is the used in the DSC controller ? I'm a little confused about if this is a DSP concern or a generat DSC concern. in the first case, the DSP controlelr should set its own condition, the DSC controller can eventually copy them.

@VaishnaviHire do you remember what is the logic here ?

Reply:
@jackdelahunt the `UnmanagedArgoWorkFlowExists` is also called here: https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/components/datasciencepipelines/datasciencepipelines.go#L108-L110

Reply:
I am kind of confused but here is what I am understanding:

- `UnmanagedArgoWorkFlowExists` can be a new action in the DSP controller
- We can also set the first condition in the DSP controller 
-  And if that condition is there we can set the component condition in the DSC controller

Reply:
correct

Reply:
New commit has those changes but I have some questions:

1. I have this at the top of the new action, in terms of upgrading does this work the same way or because now this is an action do things act a bit differently 
```
// Check preconditions if this is an upgrade
if rr.Instance.GetStatus().Phase != status.PhaseReady {
	return nil
}
```

2. I set the `CapabilityDSPv2Argo` condition for the DSC in this action which is what I think you were looking for, is updating the DSC from a component okay? Just wondering what are the rule around component/DSC interactions

3. The second part of  `SetExistingArgoCondition`  
```
status.SetComponentCondition(conditions, componentsv1.DataSciencePipelinesComponentName, status.ReconcileFailed, message, corev1.ConditionFalse)
```
I am not sure where to put this in the DSC, I think failing the new `UnmanagedArgoWorkFlowExists` action already cause the component's reconcile to be in a failed state because it is erroring. So it this already done for us when we process the `componentErrors`? Or maybe things are acting differently I am not sure

This is my first PR on the operator so my understanding is limited but hopefully growing 😃